lib/dataset/variable_reduction.cpp

```diff
@@ -9,27 +9,68 @@
 #include "scipp/variable/creation.h"
 #include "scipp/variable/reduction.h"
 #include "scipp/variable/special_values.h"
-#include "scipp/variable/transform.h"
 #include "scipp/variable/util.h"
 
 #include "dataset_operations_common.h"
 
 namespace scipp::dataset {
 
-Variable sum(const Variable &var, const Dim dim, const Masks &masks) {
+namespace {
+// Uses variable::special like but constructs only a scalar
+// instead of a full-sized array.
+Variable mask_fill(const Variable &prototype, const FillValue fill) {
+  return special_like(zero_like(prototype), fill);
+}
```

Comment:
Name is a bit confusing. `special_scalar_like`, maybe?

Reply:
Actually it is used only once, just remove it?

Reply:
Renamed it because I think that it (with the comment) helps to understand the purpose.